dll/win32/shell32/shlexec.cpp

```diff
@@ -2031,6 +2090,18 @@ static BOOL SHELL_execute(LPSHELLEXECUTEINFOW sei, SHELL_ExecuteW32 execfunc)
         return retval > 32;
     }
 
+    if (!(sei_tmp.fMask & SEE_MASK_IDLIST) && // Not an ID List
+        !PathIsShortcutFile(sei_tmp.lpFile)) // CShellLink::InvokeCommand depends on ShellExecute
```

Comment:
so basically, this is to avoid recursive calls?

Reply:
Yep.